app/cli/cmd/casbackend_add_s3.go

```diff
@@ -46,6 +48,11 @@ func newCASBackendAddAWSS3Cmd() *cobra.Command {
 				}
 			}
 
+			// If there is a custom endpoint we want to store it as part of the fqdn location
+			if endpoint != "" {
```

Comment:
the reason of storing the endpoint in the location instead of the credentials is so it's shown to the user and can't be updated later on.